recipes/ha_namenode_primary_init.rb

```diff
@@ -0,0 +1,27 @@
+#
+# Cookbook Name:: hadoop_wrapper
+# Recipe:: ha_namenode_primary_init
+#
+# Copyright © 2013-2015 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This is to ensure that we've been formatted
+include_recipe 'hadoop_wrapper::hadoop_hdfs_namenode_init'
+
+ruby_block 'initaction-hdfs-namenode-initialize-sharededits' do
+  block do
+    resources(:execute => 'hdfs-namenode-initialize-sharededits').run_action(:run)
```

Comment:
this will fail if the preceding namenode format was performed while configured against the journal nodes (quorum dir not empty error).  We should try to protect against that.  Potentially, we should only run this if the preceding ``initaction-format-namenode`` resource was skipped due to not_if.

It seems like there are multiple contexts